src/components/application_manager/test/commands/mobile/diagnostic_message_request_test.cc

```diff
@@ -0,0 +1,178 @@
+
+typedef SharedPtr<DiagnosticMessageRequest> DiagnosticMessageRequestPtr;
+
+namespace {
+const uint32_t kConnectionKey = 2u;
+const uint32_t kDiagnosticMode = 5u;
+}  // namespace
+
+class DiagnosticMessageRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {};
+
+TEST_F(DiagnosticMessageRequestTest, Run_ApplicationIsNotRegistered_UNSUCCESS) {
+  DiagnosticMessageRequestPtr command(
+      CreateCommand<DiagnosticMessageRequest>());
+
+  EXPECT_CALL(app_mngr_, application(_))
+      .WillOnce(Return(ApplicationSharedPtr()));
+
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(
+          MobileResultCodeIs(mobile_result::APPLICATION_NOT_REGISTERED), _));
+
+  command->Run();
+}
+
+TEST_F(DiagnosticMessageRequestTest, Run_NotSupportedDiagnosticMode_UNSUCCESS) {
+  MessageSharedPtr command_msg(CreateMessage(smart_objects::SmartType_Map));
+  (*command_msg)[am::strings::msg_params][am::strings::message_data][0] =
+      kDiagnosticMode;
+  (*command_msg)[am::strings::params][am::strings::connection_key] =
+      kConnectionKey;
+
+  DiagnosticMessageRequestPtr command(
+      CreateCommand<DiagnosticMessageRequest>(command_msg));
+
+  MockAppPtr app(CreateMockApp());
+  EXPECT_CALL(app_mngr_, application(kConnectionKey)).WillOnce(Return(app));
+
+  EXPECT_CALL(app_mngr_, get_settings())
+      .WillOnce(ReturnRef(app_mngr_settings_));
+
+  const std::vector<uint32_t> kEmptySupportedDiagModes;
+  EXPECT_CALL(app_mngr_settings_, supported_diag_modes())
+      .WillOnce(ReturnRef(kEmptySupportedDiagModes));
+
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(MobileResultCodeIs(mobile_result::REJECTED), _));
+
+  command->Run();
+}
+
+TEST_F(DiagnosticMessageRequestTest, Run_SUCCESS) {
+  MessageSharedPtr command_msg(CreateMessage(smart_objects::SmartType_Map));
+  (*command_msg)[am::strings::msg_params][am::strings::message_data][0] =
+      kDiagnosticMode;
+  (*command_msg)[am::strings::params][am::strings::connection_key] =
+      kConnectionKey;
+
+  DiagnosticMessageRequestPtr command(
+      CreateCommand<DiagnosticMessageRequest>(command_msg));
+
+  MockAppPtr app(CreateMockApp());
+  EXPECT_CALL(app_mngr_, application(kConnectionKey)).WillOnce(Return(app));
+
+  EXPECT_CALL(app_mngr_, get_settings())
+      .WillOnce(ReturnRef(app_mngr_settings_));
+
+  std::vector<uint32_t> supported_diag_modes;
+  supported_diag_modes.push_back(kDiagnosticMode);
+
+  EXPECT_CALL(app_mngr_settings_, supported_diag_modes())
+      .WillOnce(ReturnRef(supported_diag_modes));
+
+  EXPECT_CALL(app_mngr_,
+              ManageHMICommand(HMIResultCodeIs(
+                  hmi_apis::FunctionID::VehicleInfo_DiagnosticMessage)));
+
+  command->Run();
+}
+
+TEST_F(DiagnosticMessageRequestTest, OnEvent_UNSUCCESS) {
+  Event event(hmi_apis::FunctionID::INVALID_ENUM);
+
+  DiagnosticMessageRequestPtr command(
+      CreateCommand<DiagnosticMessageRequest>());
+
+  EXPECT_CALL(app_mngr_, ManageMobileCommand(_, _)).Times(0);
```

Comment:
@LuxoftAKutsan, In this case - no. Because `DiagnosticMessageRequest` should not to call `ManageMobileCommand(..)` in any case.